From ad28b585c37346abb9988898d52d08ca70a3f195 Mon Sep 17 00:00:00 2001
From: Lei Wen <leiwen@marvell.com>
Date: Sat, 23 May 2009 16:51:30 +0800
Subject: [PATCH] pxa168: change mtd page and size calculation way

The old calculation way is not applied to K9LBG08U0D,
so change the original which follow the nand_base.

The code in nand_base is also need to be changed, but is not
under our control.

Signed-off-by: Lei Wen <leiwen@marvell.com>
---
 drivers/mtd/nand/pxa3xx_nand.c |   17 +++--------------
 1 files changed, 3 insertions(+), 14 deletions(-)

diff --git a/drivers/mtd/nand/pxa3xx_nand.c b/drivers/mtd/nand/pxa3xx_nand.c
index 1ab110b..7c3212b 100644
--- a/drivers/mtd/nand/pxa3xx_nand.c
+++ b/drivers/mtd/nand/pxa3xx_nand.c
@@ -996,20 +996,9 @@ static int pxa3xx_nand_detect_flash(struct pxa3xx_nand *nand)
 			if (f->chip_id == id) {
 				pxa3xx_nand_config_flash(info, f);
 				chip->cellinfo = info->data_buff[2];
-				extid = info->data_buff[3];
-				/* Calc pagesize */
-				mtd->writesize = 1024 << (extid & 0x3);
-				extid >>= 2;
-				/* Calc oobsize */
-				mtd->oobsize = (8 << (extid & 0x01))    \
-					       * (mtd->writesize >> 9);
-
-				extid >>= 2;
-
-				/* Calc blocksize. Blocksize is multiples */
-				/* of 64KiB 				  */
-				mtd->erasesize = (64 * 1024) 	    \
-						 << (extid & 0x03);
+				mtd->writesize = f->page_size;
+				mtd->oobsize = mtd->writesize / 32;
+				mtd->erasesize = f->page_size * f->page_per_block;
 
 				mtd->name = f->name;
 				break;
-- 
1.6.0.4

